src/StaticWebAssetsSdk/Tasks/WriteImportMapToHtml.cs

```diff
@@ -0,0 +1,204 @@
+using System.Text.Json;
+
+namespace Microsoft.AspNetCore.StaticWebAssets.Tasks;
+
+public partial class WriteImportMapToHtml : Task
+{
+    [Required]
+    public ITaskItem[] Endpoints { get; set; } = [];
+
+    [Required]
+    public string OutputPath { get; set; } = string.Empty;
+
+    [Required]
+    public ITaskItem[] HtmlFiles { get; set; } = [];
+
+    [Output]
+    public ITaskItem[] HtmlCandidates { get; set; } = [];
+
+    [Output]
+    public ITaskItem[] HtmlFilesToRemove { get; set; } = [];
+
+    [Output]
+    public string[] FileWrites { get; set; } = [];
+
+    // "([^"]+)(#\[\.{fingerprint}\])([^"]+)"
+    // 1.group = file name
+    // 2.group = fingerprint placeholder
+    // 3.group = file extension
+    // wrapped in quotes
+    private static readonly Regex _assetsRegex = new Regex(@"""([^""]+)(#\[\.{fingerprint}\])([^""]+)""");
+
+    private static readonly Regex _importMapRegex = new Regex(@"<script\s+type=""importmap""\s*>\s*</script>");
+
+    public override bool Execute()
+    {
+        var endpoints = StaticWebAssetEndpoint.FromItemGroup(Endpoints).Where(e => e.AssetFile.EndsWith(".js") || e.AssetFile.EndsWith(".mjs"));
+        var resources = CreateResourcesFromEndpoints(endpoints);
+        var urlMappings = GroupResourcesByLabel(resources);
+        var importMap = CreateImportMapFromResources(resources);
+
+        var htmlFilesToRemove = new List<ITaskItem>();
+        var htmlCandidates = new List<ITaskItem>();
+        var fileWrites = new List<string>();
+
+        if (!Directory.Exists(OutputPath))
+        {
+            Directory.CreateDirectory(OutputPath);
+        }
+
+        foreach (var item in HtmlFiles)
+        {
+            if (File.Exists(item.ItemSpec))
+            {
+                string content = File.ReadAllText(item.ItemSpec);
+
+                // Generate import map
+                string outputContent = _importMapRegex.Replace(content, e =>
+                {
+                    Log.LogMessage("Writing importmap to '{0}'", item.ItemSpec);
+                    return $"<script type=\"importmap\">{JsonSerializer.Serialize(importMap, ImportMapSerializerContext.CustomEncoder.Options)}</script>";
+                });
+
+                // Fingerprint all assets used in html
+                outputContent = _assetsRegex.Replace(outputContent, e =>
+                {
+                    string assetPath = e.Groups[1].Value + e.Groups[3].Value;
+                    string fingerprintedAssetPath = urlMappings.TryGetValue(assetPath, out var value) ? value.Url : assetPath;
+                    Log.LogMessage("Replacing asset '{0}' with fingerprinted version '{1}'", assetPath, fingerprintedAssetPath);
+                    return "\"" + fingerprintedAssetPath + "\"";
+                });
+
+                if (content != outputContent)
+                {
+                    string outputPath = Path.Combine(OutputPath, Path.GetRandomFileName() + item.GetMetadata("Extension"));
```

Comment:
Don't do this. Use a hash of the content if anything, but don't use `Random` as this makes things non-deterministic

Reply:
Also use https://github.com/dotnet/sdk/blob/main/src/StaticWebAssetsSdk/Tasks/Utils/ArtifactWriter.cs